### ISSUE TITLE
HUD and Skill System

### ISSUE THREAD

**Issue by ghost** [PlaceholderGames/2016-VnD_Game#9]
This is where all issues relating to the HUD (Head-Up Display) will be kept, the HUD will display information to the player including the total ants, number of ants remaining, time remaining, the mini-map, and limit of skill uses and also has clickable skill buttons and shows the players current target.

PNG format icons will need to be designed for each individual skill, these must match the theme of the game.

Assigned to: Richard Davies

**Comment by ghost** [comment]
Here's my initial concept for the players HUD and click-able skill buttons, I needed a visual image before I start writing the code.

Without a target selected the player's HUD will be clean only showing the time remaining on the level and the amount of ants on the level and how many left to complete the level. Its not decided if all levels will have a time limit but it seems a logical choice at the moment.

![Players HUD without target](https://cloud.githubusercontent.com/assets/22819171/19975282/49be1d5e-a1e3-11e6-8391-94b6f5d5c76b.PNG)

Once a player has clicked an Ant, it becomes the players target the HUD shows this.

Each ant in the array will be assigned a number and a randomise name from a predefined list. The click-able skills will be displayed on the right of the ants head which can be seen below, these could potentially be assigned hotkeys as well.

![Players HUD with target](https://cloud.githubusercontent.com/assets/22819171/19975296/55ae6894-a1e3-11e6-9575-65eb49cf0461.PNG)

**Comment by ghost** [comment]
Work in progress:

Attached the HUD to the screen, Now to work on making the HUD operate correctly as I have described above, I've also changed the camera to 1st Person on my branch of the project I'll work on the camera there so it feels like a god travelling above his realm.

![capture](https://cloud.githubusercontent.com/assets/22819171/19975111/ad51b944-a1e2-11e6-9aa2-41594f17303c.PNG)

I also based the sprites of the HUD on a screen size of 1920 x 1080 so its easy to downscale them later on, because up-scaling doesn't work so well.  These are just placeholders at the moment.

**Comment by ghost** [comment]
@DoctorMikeReddy To this point its a duplicate of the Basecamp thread, From now on all the updates will be written here, Instead of basecamp.
DoctorMike: Noted

**Comment by ghost** [comment]
Timer and current_ants / max_ants working and displaying on HUD Correctly, Now for the Ants name and number to be displayed when targeting an ant!

![capture](https://cloud.githubusercontent.com/assets/22819171/19991184/5e63ddd2-a22b-11e6-978b-76a8369e82ba.PNG)

**Comment by ghost** [comment]
Here's the updated HUD, the graphics are currently being made for the portrait outline, ant typea, hotkey icons and the stopwatch.

![hud](https://cloud.githubusercontent.com/assets/22819171/22188945/2aa974a0-e111-11e6-8450-f25db247031a.PNG)

When you click F1 the hotkey image goes blank and the cool-down timer shows the hotkey icon then returns when the skills isn't on cool-down anymore, I plan to do further work and try and code a cool-down animation behind the timer in the hotkey, This can be seen on many MMO's.

![hud on cooldown](https://cloud.githubusercontent.com/assets/22819171/22189009/9f8abc3e-e111-11e6-8ad2-4a0e360940f1.PNG)

Also no bugs possible I've error tested it for ages, It'll look really polished when the graphics are done!

Also I'm open to suggestions on the font style from you guys? :)

**Comment by BojanStankovic** [comment]
Hi @Richard-Chaos,

Great work on the HUD so far. Would it be possible to finish this issue without any polishing of the graphics at this point? As we discussed in J208 last Thursday, this is not important for the first build, but the basic functionality is. Also, any new features like cool-down are a welcome addition but not crucial. Please, just wrap up the basic HUD so the rest of the team can start using it since it is needed for other issues. When you are done push it to the repository on your branch so I can merge it to the master and mark it here as ready for reviewing and merging.

A small request for future project workflow:
- use the naming convention for the branches instead using something like "RichardsBranch"
- one branch should correspond to one issue (applicable in most cases, even with bugs and improvements)
- one issue should be one small logical task/feature/fix

Regarding the last point your HUD branch is already having few tasks that could have been separated in several issues so any new feature should be a new issue. Please feel free to ask for any clarification about this workflow approach, I'll be happy to help. Also, let me know if you need help pushing your updates to the repo :smiley:

**Comment by ghost** [comment]
I've deleted my old branch "Richard's_Branch" and created a new branch using the correct naming conventions called "I9_HUD_and_Skill_System". Its been pushed, up to date and working.  I've tested it thoroughly for errors so hopefully its fine! Still waiting on the graphics but it should do for the moment.

**Comment by ghost** [comment]
Its ready for review now.

**Comment by BojanStankovic** [comment]
It seems like all is there. I found few things that need fixing, but we'll do that as we go. I will merge it to the develop branch and report any new issues. Closing...
